### DISCUSSION TITLE
Switch UUID to ULID

### DISCUSSION THREAD

**Discussion by lafriks** [woodpecker-ci/woodpecker#2156] [Request a Feature]
For tasks etc when generating a unique ID it would be better to switch to using ULID as unique IDs as they are shorter, contain timestamp encoded in them and are sortable

**Comment by 6543** [comment]
we already store timestamps so that's not what we need. the question would be, how many times shorter it is and how much resources we save with it

**Comment by lafriks** [comment]
It's not so much about storage space per-se but more about readability etc in docker monitoring and sorting:

```
01AN4Z07BY79KA1307SR9X4MV3
vs
3A7DF383-F8C5-4679-A3F4-CA50CAAAA7E1
```

**Comment by 6543** [comment]
Well asking few other people ... uuid is mor redable becouse of the defider `-` 😆

**Comment by lafriks** [comment]
I did not mean readable in that kind of sense as no one would read UUID itself but step names are more readable in container list because of not having dashes in them

**Comment by 6543** [comment]
Well I'm not against or for a specific ID generator ... as long as it does not create colisions and use well known libs I wont block it :)

**Comment by lafriks** [comment]
I have been using ULID for quite some time using `github.com/oklog/ulid`.
ULID's are UUID compatible and are 128-bit long
